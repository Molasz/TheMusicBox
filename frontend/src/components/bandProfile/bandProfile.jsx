import React, { useEffect } from 'react';
import { connect } from 'react-redux';

import './bandProfile.scss';

import { getBand } from '../../redux/actions/bandActions';
import store from '../../redux/store';

import Photos from './photos/photos';
import Discography from './discography/discography';
import Concerts from './concerts/concerts';
import Bio from './bio/bio';

import GradeIcon from '@material-ui/icons/Grade';

function BandProfile(props) {
  useEffect(() => {
    store.dispatch(getBand(props.match.params.bandId));
  }, []);

  const result = props.band ? (
    <article className='band-profile'>
      <div className='band-profile__top'>
        <img src={props.band.logo} alt='Logo' className='top__logo' />

        <img
          src='https://pbs.twimg.com/profile_banners/851494228815482880/1539011178/1500x500'
          alt='Banner'
          className='top__banner'
        />
        <div className='top__info'>
          <strong className='info__name'>{props.band.name}</strong>
          <div className='info__follow'>
            <GradeIcon />
            <button className='follow__button'>Follow</button>
            <p className='follow__number'> 14 Followers</p>
          </div>
        </div>
      </div>
      <div className='band-profile__middle'>
        <div className='middle__bio'>
          <Bio
            bio={props.band.bio}
            city={props.band.city}
            country={props.band.country}
            name={props.band.name}
          />
        </div>

        <Discography data={props.band.discography} />
      </div>
      <div className='band-profile__bottom'>
        <div className='bottom__concerts'>
          <Concerts data={props.band.concerts} />
        </div>
        <div className='bottom__photos'>
          <Photos data={props.band.photos} />
        </div>
      </div>
    </article>
  ) : (
    <h3>Loading...</h3>
  );
  return result;
}

function mapStateToProps(state) {
  return state.bandReducer;
}

function mapDispatchToProps(dispatch) {
  return { getBand };
}
export default connect(mapStateToProps, mapDispatchToProps)(BandProfile);
