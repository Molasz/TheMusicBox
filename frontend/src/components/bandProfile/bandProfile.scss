$color-detail: #f55110;
.band-profile {
  min-width: 500px;
  display: flex;
  flex-direction: column;

  color: white;
  padding: 20px 10%;
  .band-profile__top {
    display: flex;
    flex-direction: row;

    padding-bottom: 13px;
    border-bottom: 4px solid $color-detail;
    margin-bottom: 0px;
    .top__logo {
      width: 50%;
    }
    .top__banner {
      display: none;
      width: 50%;

      margin-left: 30px;
    }
    .top__info {
      display: flex;
      flex-direction: column;
      justify-content: space-between;

      width: 30%;
      margin-left: 30px;

      .info__name {
        text-transform: uppercase;
        font-size: 80px;
        margin: 0;
      }

      .info__follow {
        .follow__container {
          display: flex;
          align-items: center;

          .contanier__icon {
            font-size: 30px;
            margin-right: 10px;
          }

          .white {
            color: white;
          }

          .orange {
            color: $color-detail;
          }
        }
      }
    }
  }
  .band-profile__middle {
    display: flex;
    flex-direction: column;

    padding-bottom: 20px;
    border-bottom: 4px solid $color-detail;
  }
  .band-profile__bottom {
    display: flex;
    flex-direction: column;
  }
}

// Medium devices (tablets, 768px and up)
@media (min-width: 768px) {
}

// Large devices (desktops, 992px and up)
@media (min-width: 992px) {
  .band-profile {
    .band-profile__top {
      .top__logo {
        width: 20%;
      }
      .top__banner {
        display: block;
      }
      .top__info {
        display: flex;
        flex-direction: column;

        margin-left: 30px;
        .info__name {
          font-size: 80px;
        }
      }
    }
    .band-profile__middle {
      flex-direction: row;
    }

    .band-profile__bottom {
      flex-direction: row;
    }
  }
}
