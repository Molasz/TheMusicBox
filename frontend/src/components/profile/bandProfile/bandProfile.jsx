import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';

import './bandProfile.scss';

import PuffLoader from 'react-spinners/PuffLoader';

import { getBand, follow, showDisc } from '../../../redux/actions/bandActions';

import ProfileHeader from '../profileHeader/profileHeader';
import Photos from './photos/photos';
import Discography from './discography/discography';
import Concerts from './concerts/concerts';
import Bio from '../bio/bio';

function BandProfile({ match, band, followers, user, dispatch }) {
  const [calls, setCalls] = useState(false);
  const [checkFollowing, setCheckFollowing] = useState(false);
  const [isFollowing, setIsFollowing] = useState(null);

  useEffect(() => {
    if (!calls) {
      dispatch(showDisc());
      dispatch(getBand(match.params.bandId));
      dispatch(follow(match.params.bandId));
      setCalls(true);
    }
    if (
      !checkFollowing &&
      !(Object.keys(user).length === 0 && user.constructor === Object) &&
      !(Object.keys(band).length === 0 && band.constructor === Object)
    ) {
      setIsFollowing(
        user.following.some((element) => element._id === band._id)
      );
      setCheckFollowing(true);
    }
  }, [band, calls, dispatch, match.params.bandId, user, checkFollowing]);

  const result = !(
    Object.keys(band).length === 0 && band.constructor === Object
  ) ? (
    <article className='band-profile'>
      <div className='band-profile__top'>
        <ProfileHeader
          logo={band.logo}
          banner={band.banner}
          name={band.name}
          isBand={true}
          match={match}
        />
      </div>
      <div className='band-profile__middle'>
        <Bio
          bio={band.bio}
          city={band.city}
          country={band.country}
          name={band.name}
          socialNetwork={band.socialNetwork}
        />

        <Discography data={band.discography} />
      </div>
      <div className='band-profile__bottom'>
        <Concerts data={band.concerts} />
        <Photos data={band.photos} />
      </div>
    </article>
  ) : (
    <PuffLoader color='#f55110' />
  );

  return result;
}

function mapStateToProps(state) {
  return {
    band: state.bandReducer.band,
    user: state.authReducer.user
  };
}

export default connect(mapStateToProps, null)(BandProfile);
export { BandProfile };
