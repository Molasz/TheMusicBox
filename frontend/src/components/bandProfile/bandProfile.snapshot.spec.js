import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { BandProfile } from './bandProfile';

describe('Search snapshot', () => {
  xit('Should match with band data', () => {
    const band = {
      _id: 0,
      logo: 'logo',
      banner: 'banner',
      name: 'name',
      bio: 'bio',
      city: 'city',
      country: 'country',
      socialNetwork: ['twitter'],
      discography: [
        {
          title: 'title',
          img: 'img',
          date: 'date',
          songs: [
            {
              title: 'title',
              time: 'time'
            }
          ]
        }
      ],
      concerts: [{ date: 'date', city: 'city' }],
      photos: ['photo']
    };
    const tree = renderer.create(
      <BrowserRouter>
        <BandProfile band={band} />
      </BrowserRouter>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Should match without band data', () => {
    // const followers = 0;
    // const user = { _id: 0 };
    const tree = renderer.create(
      <BrowserRouter>
        <BandProfile band={null} /*followers={followers} user={user}*/ />
      </BrowserRouter>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
