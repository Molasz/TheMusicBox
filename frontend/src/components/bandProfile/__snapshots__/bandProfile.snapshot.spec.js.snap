// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BandProfile snapshot Should match with data 1`] = `
<article
  className="band-profile"
>
  <div
    className="band-profile__top"
  >
    <img
      alt="Logo"
      className="top__logo"
      src="logo"
    />
    <img
      alt="Banner"
      className="top__banner"
      src="banner"
    />
    <div
      className="top__info"
    >
      <strong
        className="info__name"
      >
        name
      </strong>
      <div
        className="info__follow"
      >
        <div
          className="follow__container"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root contanier__icon white"
            focusable="false"
            id="follow"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
          <p
            className="follow__count"
          >
             
             Followers
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="band-profile__middle"
  >
    <section
      className="bio"
    >
      <div
        className="bio__top"
      >
        <p
          className="top__title"
        >
          <strong
            className="title__name"
          >
            name
          </strong>
           | 
          <span
            className="title__city"
          >
            city
          </span>
             
          <span
            className="title__country"
          >
            country
          </span>
        </p>
        <p
          className="top__text"
        >
          bio
        </p>
      </div>
      <div
        className="bio__bottom"
      >
        <a
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root bottom-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
            />
          </svg>
        </a>
        <a
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root bottom-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
            />
          </svg>
        </a>
        <a
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root bottom-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
            />
          </svg>
        </a>
      </div>
    </section>
  </div>
  <div
    className="band-profile__bottom"
  >
    <section
      className="concerts"
    >
      <div
        className="concerts__title"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root title__icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 3v10.55c-.59-.34-1.27-.55-2-.55-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4V7h4V3h-6z"
          />
        </svg>
        <h1
          className="title__text"
        >
          Concerts
        </h1>
      </div>
      <div
        className="concerts__main"
      >
        <div
          className="main__item"
        >
          <p
            className="item__date"
          />
          <p
            className="item__city"
          />
        </div>
      </div>
    </section>
    <section
      className="photo"
    >
      <div
        className="photo__title"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root title__icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22 16V4c0-1.1-.9-2-2-2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2zm-11-4l2.03 2.71L16 11l4 5H8l3-4zM2 6v14c0 1.1.9 2 2 2h14v-2H4V6H2z"
          />
        </svg>
        <h1
          className="title__text"
        >
          Photos and videos
        </h1>
      </div>
      <div
        className="photo__main"
      >
        <img
          alt="nÂº0"
          className="main__item"
          src="photo"
        />
      </div>
    </section>
  </div>
</article>
`;

exports[`BandProfile snapshot Should match without data 1`] = `
<h3>
  Loading...
</h3>
`;
